src/hash.h

```diff
@@ -200,6 +213,9 @@ uint256 SerializeHash(const T& obj, int nType=SER_GETHASH, int nVersion=PROTOCOL
     return ss.GetHash();
 }
 
+/** Single-SHA256 a 32-byte input (represented as uint256). */
+uint256 SHA256Uint256(const uint256& input);
```

Comment:
Perhaps add `NODISCARD` here to guard against the issue you pointed out in the other PR.